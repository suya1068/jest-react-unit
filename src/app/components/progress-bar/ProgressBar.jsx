import React from "react";

export default function ProgressBar() {
    return (
        <div className="u-progress-bar">
            ProgressBar
        </div>
    );
}
