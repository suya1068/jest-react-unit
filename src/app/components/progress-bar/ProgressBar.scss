.u-progress {
    position: relative;
    display: block;
    padding: 0;
    margin: 1em 0 2.5em;
    max-width: 100%;
    background-color: rgba(0, 0, 0, 0.1);
    border: none;
    border-radius: 0.25rem;

    &__bar {
        display: block;
        height: 1.75em;
        line-height: 1;
        border: none;
        border-radius: 0.25rem;
        transition: width, background-color 0.1s ease;
    }

    &__label {
        position: absolute;
        width: 100%;
        left: 0;
        top: 100%;
        right: auto;
        bottom: auto;
        color: rgba(0, 0, 0, 0.87);
        font-size: 1em;
        font-weight: 700;
        margin-top: 0.2em;
        text-shadow: none;
        text-align: center;
        transition: color 0.4s ease;
    }
}
