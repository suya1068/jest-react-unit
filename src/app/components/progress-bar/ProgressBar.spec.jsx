import React from "react";
import ProgressBar from "./ProgressBar";

describe("# ProgressBar Component", () => {
    it("render <ProgresBar />", () => {
        const tree = renderer
            .create(<ProgressBar />)
            .toJSON();
            
        expect(tree).toMatchSnapshot();
    });
});
