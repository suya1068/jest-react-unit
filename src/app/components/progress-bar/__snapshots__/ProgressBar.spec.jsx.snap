// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# ProgressBar Component render <ProgresBar /> 1`] = `
<div
  className="u-progress-bar"
>
  ProgressBar
</div>
`;
